### ISSUE TITLE
Missing Fromalert and X-Cache-Skip HTTP headers since v9.1.0

### ISSUE THREAD

**Issue by camathieu** [grafana/grafana#54403]
**What happened**:

After upgrading to v9.1.1 we noticed that the `Fromalert` header we use to route Prometheus alerting queries was not added to the requests anymore.

Grafana 9.0.7 :
```
POST /prometheus/api/v1/query_range HTTP/1.1
Host: 127.0.0.1:8427
Accept-Encoding: gzip
Content-Length: 62
Content-Type: application/x-www-form-urlencoded
Fromalert: true
User-Agent: Grafana/9.0.7
X-Cache-Skip: true
```

Grafana 9.1.0
```
POST /prometheus/api/v1/query_range HTTP/1.1
Host: 127.0.0.1:8427
Accept-Encoding: gzip
Content-Length: 62
Content-Type: application/x-www-form-urlencoded
User-Agent: Grafana/9.1.0
```

**What you expected to happen**:

Fromalert and X-Cache-Skip Headers are added to alerting queries requests

**How to reproduce it (as minimally and precisely as possible)**:

Upgrade from Grafana 9.0.x to Grafana 9.1.x

**Anything else we need to know?**:

Might be related to this commit : https://github.com/grafana/grafana/commit/3df34fe06452d7a5c5369b27d81514497a57485b

**Comment by kylebrandt** [comment]
Looks to me this was initially removed in https://github.com/grafana/grafana/pull/51518 , based on the assumption that the headers were internal only.

**Comment by camathieu** [comment]
Would you consider adding that header back ? Especially the `fromalert : true` ?

**Comment by josiahg** [comment]
Hi @camathieu ! Thanks for highlighting this. I'm sorry for the frustration this change caused.

Can you share a bit with us about what you're using these headers for? We're looking for the best way to support your use case.

**Comment by camathieu** [comment]
Hey @josiahg,

I sure can. We use this header for two things : 
1 - To ensure that users can only run alerting queries over metrics with a guaranteed SLA.
2 - To ensure that users do not overload our backends with huge alerting queries.

1) We manage more than half a billion metrics distributed in a wide range of backends each with different retention and SLAs. Some of those backend are not suited for alerting queries but only for troubleshooting purposes and don't provide any availability/retention guarantees. We use this header to exclude those backend for alerting queries to avoid our users to setup critical page alerts using those, forcing them to write the adequate recording rules.

2) Our alerting queries are routed to queriers with a lower `--query.max-samples`. That allow our user to run _big_ queries in explore and dashboards and force them to run only carefully crafted queries for recurring alerting queries which represents by far the most of our read workload.

3) As the alerting queries are so crucial for user to be confident that pages will work when they need to we did setup a specific monitoring of those queries and users can be alerted when a specific query fails or timeout too much. We can then help them improving the quality of those query. We also periodically review the slowest, most flaky alerting queries.

This header is really helpful in this regard as managing a separate set of datasources for alerting would not really be helping much.

One small thing we noticed is that the `fromalert` header is present in the alerting panel query editor but not in the metric browser.
